### PR TITLE
Enable Error only Properties on Telemetry logger

### LINKED ISSUE #5295: Consistent logging of op-related properties to ease debugging via telemetry
Most errors coming from the runtime should log the current sequence number and other related properties that will help us rebuild a timeline when debugging from telemetry.  We can consider stamping these on non-error logs too but that may be too heavy.

These properties are logged here and there, sometimes with varying names, etc, so consolidating names and ensuring they're logged pervasively will make a big difference.

Let's document these and consider it almost like an API design that we somewhat "lock down".  And put under test via mock logger?

### PR DESCRIPTION
**Pull request by anthony-murphy**
This is related to #5295. This change doesn't add an logging of error only properties, but in the next phase we'll start setting error properties at the container level, so all errors have a consistent set of properties

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1486,7 +1488,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 await this.refreshLatestSummaryAck(
                     undefined,
                     version.id,
-                    new ChildLogger(summaryLogger, undefined, { safeSummary: true }),
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1489]: Should `ChildLogger`'s constructor be private?

**Review by markfields — COMMENTED** [review]
🎂 LGTM! Except I really would like `ITelemetryLoggerProperty` renamed to `ITelemetryLoggerPropertyBag` and likewise the variable names of that type.

**Review by markfields — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1489]: i do exactly that below!

File: packages/utils/telemetry-utils/src/logger.ts [hunk 2]
```diff
@@ -211,27 +214,35 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
...
+                properties.push(this.properties.error);
+            }
+            for(const props of properties) {
+                if(props !== undefined) {
+                    for (const key of Object.keys(props)) {
+                        if (event[key] !== undefined) {
+                            continue;
+                        }
+                        const getterOrValue = props[key];
+                        // If this throws, hopefully it is handled elsewhere
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/logger.ts:237]: been working so far :)

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1489]: 🧐

**Pull request merged**